DTAdmin/Subjects/SubjectTableViewController.swift

```diff
@@ -0,0 +1,134 @@
+    
+    let queryService = QueryService()
+    
+    var filteredData = [Records]()
+    
+    var inSearchMode = false
+    
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        showRecords()
+        // searchBar.returnKeyType = UIReturnKeyType.done
+        searchBar.showsScopeBar = true
+        searchBar.scopeButtonTitles = ["Name","Description"]
+        searchBar.selectedScopeButtonIndex = 0
+    }
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        if searchBar.text == nil || searchBar.text == "" {
+            inSearchMode = false
+            view.endEditing(true)
+            tableView.reloadData()
+        } else {
+            inSearchMode = true
+            switch  searchBar.selectedScopeButtonIndex {
+            case 0:
+                filteredData = records.filter{$0.name.contains(searchBar.text!)}
+            case 1:
+                filteredData = records.filter{$0.description.contains(searchBar.text!)}
+                //            case 2:
+            //                filteredData = records.filter{$0.id.contains(searchBar.text!)}
+            default:
+                print("No match")
+            }
+            tableView.reloadData()
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        showRecords()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    private func showRecords() {
+        Records.getRecords(sufix: "getRecords", completion: { (results:[Records]?) in
+            if let subjectData = results {
+                self.records = subjectData
+                self.records.sort { (rec1, rec2) -> Bool in
+                    return rec1.name < rec2.name
+                }
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        })
+    }
+    
+    @IBAction func updateRecords(_ sender: UIBarButtonItem) {
+        tableView.reloadData()
+    }
+    
+    // MARK: - Table view data source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        if inSearchMode {
+            return filteredData.count
+        }
+        return records.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+        if inSearchMode {
+            cell.textLabel?.text = filteredData[indexPath.row].name
```

Comment:
IMO it's better to introduce local variable for cell data. (e.g. `let cellData = filteredData[indexPath.row]` and then use it below.